### ISSUE TITLE
Maintenance Mode Whitelist IP

### ISSUE THREAD

**Issue by nesoor** [pojome/elementor#2687]
The Maintenance mode is great but I constantly end up installing a different one.
It would be nice if there was an option to white list IP addresses.

This would make it perfect :)

**Comment by stickypages** [comment]
Yeah I agree, I think it would need to be more than just an IP though. Asking your client to provide your IP is not an easy solution. You need to just send them a link and it should work.

This is what I came up with. Fairly basic, and you could alter SESSION to be a COOKIE or THEMEMOD etc. 
I use this in my custom theme I built, and works pretty good, no issues yet.

`function sticky_maintenance_mode(){
	if(current_user_can('administrator') || is_user_logged_in() || sticky_has_access_via_url()){
		// They can view the site
	} else {
		wp_die('<h1 style="color:#666666; font-family:Arial, Helvetica, sans-serif;">Under Maintenance</h1><br />We will be back as soon as we can!');
	}
}
function sticky_has_access_via_url() {
	if(isset($_GET) && isset($_GET['key']) && !empty($_GET['key']) ) {
		$_SESSION['key'] = $_GET['key'];
	}
       // CHANGE YOUR KEY HERE
	if( $sticky_maintenance_access_key = get_theme_mod('sticky_maintenance_access_key') ) {
		if($_SESSION['key'] === $sticky_maintenance_access_key) {
			return true;
		}
	}
	return false;
}
if( get_theme_mod('sticky_enable_maintenance_mode')  ) {
	if (!session_id()) {
		session_start();
	}
	add_action('get_header', 'sticky_maintenance_mode');
}`

**Comment by nesoor** [comment]
My clients do not have that much problems with sending their ip address but sending just a link would be even easier ! ! I think what stickypages said would be a good idea.

**Comment by benedettogit** [comment]
If the white list IP is added, (which i think it is great because i have the same problem as Nesoor, and no problem asking the IP to the client, I just ask them to search my ip in google and it comes out) I would also like to have the option exclude specific pages.
The link idea is a good one, but once I send the link out the client will always be able to see the page, would i have the option to disable it if I don't want them to see it anymore, with the IP I could just remove it and i'm done.

**Comment by stickypages** [comment]
You could easily add one extra param, to check if the cookie is 'valid' after you change it.

**Comment by completeinnovation** [comment]
Any updates on this - still needing this feature for internal and beta testing (ie. membership websites where the site needs to be functional to test sign up processes etc).

**Comment by CKENT0369** [comment]
New user with elementor Pro, Really need the IP whitelist for testing before launch. Like terry said needing to test registration becomes a pain. Please enable an ip whitelist for the coming soon or maintenance modes in addition to the roles.

**Comment by RubenFixit** [comment]
It would also be very useful to whitelist pages and posts from coming soon pages.

**Comment by CharlieEtienne** [comment]
For those who want to be able to whitelist some IPs in Elementor Maintenance Mode, here is a short snippet you can paste to your theme's `functions.php`:

```php
function my_elementor_maintenance_mode_whitelist_ips() {

	// IPs Whitelist
	$whitelist = [
		'127.0.0.1',
		'88.123.181.218',
	];

	if (
		in_array( $_SERVER[ 'REMOTE_ADDR' ], $whitelist )
		&& method_exists( '\Elementor\Plugin', 'instance' )
		&& isset( \Elementor\Plugin::instance()->maintenance_mode )
	) {
		remove_action( 'template_redirect', [ \Elementor\Plugin::instance()->maintenance_mode, 'template_redirect' ], 11 );
	}
}
add_action( 'template_redirect', 'my_elementor_maintenance_mode_whitelist_ips', 5 );
```

Enjoy!

**Comment by ThaDaVos** [comment]
The following issue has been closed and this one seems to be open for 7 years now - but when can we expect timer functionality? So auto-turn-off?

https://github.com/elementor/elementor/issues/5829
